### PR TITLE
[video_player_avplay] Replace g_idle_add with ecore_pipe

### PR DESCRIPTION
**Pull request by xiaowei-guan**
1. Use the same type of interface when send event to main loop. 
2. Destory event channel first when destory player.

### CODE REVIEW

File: packages/video_player_avplay/tizen/src/drm_manager.cc [hunk 1]
```diff
@@ -34,11 +34,19 @@ DrmManager::DrmManager() : drm_type_(DM_TYPE_NONE) {
...
+      },
+      this);
 }
 
 DrmManager::~DrmManager() {
   ReleaseDrmSession();
-
+  if (license_request_pipe_) {
+    ecore_pipe_del(license_request_pipe_);
+  }
```
Comment (hyue7) [inline packages/video_player_avplay/tizen/src/drm_manager.cc:49]:
```suggestion
    ecore_pipe_del(license_request_pipe_);
    license_request_pipe_ = nullptr;
  }
```

File: packages/video_player_avplay/tizen/src/media_player.cc [hunk 2]
```diff
@@ -52,7 +53,22 @@ MediaPlayer::MediaPlayer(flutter::BinaryMessenger *messenger,
...
+    player_stop(player_);
+    player_unprepare(player_);
+    player_unset_buffering_cb(player_);
+    player_unset_completed_cb(player_);
+    player_unset_interrupted_cb(player_);
+    player_unset_error_cb(player_);
+    player_unset_subtitle_updated_cb(player_);
+    player_destroy(player_);
+    player_ = nullptr;
+  }
```
Comment (JSUYA) [inline packages/video_player_avplay/tizen/src/media_player.cc:67]: How about making `PlusPlayer`'s destructor code style and error log the same?
(`drm_manager_` and `player_` seem to have the same life cycle, but the processing results vary if player_ is null in `MediaPlayer` class.)

**Review by JSUYA — APPROVED** [review]

File: packages/video_player_avplay/tizen/src/media_player.cc (hunk 2 above)
Comment (xiaowei-guan) [reply packages/video_player_avplay/tizen/src/media_player.cc:67]: Fixed

File: packages/video_player_avplay/tizen/src/drm_manager.cc (hunk 1 above)
Comment (xiaowei-guan) [reply packages/video_player_avplay/tizen/src/drm_manager.cc]: Fixed

**Pull request merged**